infra/general/iniUtils.php

```diff
@@ -7,21 +7,35 @@ class iniUtils
 {
 	/**
 	 * Given an associative array, this function will generate INI file string that represent it.
-	 * @param array $iniData
+	 * @param $ini
+	 * @param bool $isBaseLevel
+	 * @param null $baseKey
 	 * @return string
 	 */
-	public static function arrayToIniString(array $iniData)
+	public static function arrayToIniString($ini, $isBaseLevel = true, $baseKey = null)
 	{
 		$res = '';
-		foreach ($iniData as $key => $value)
+		foreach ($ini as $key => $value)
 		{
-			if (is_array($value))
+			if (!is_array($value))
 			{
-				$res .= "\n[$key]\n" . self::arrayToIniString($value);
+				$levelKey = $baseKey ? $baseKey . ".$key" : $key;
+				$res .= $levelKey . " = " . (is_numeric($value) ? $value : '"' . $value . '"') . "\n";
 			}
-			else
+		}
+		foreach ($ini as $key => $value)
+		{
+			if (is_array($value))
 			{
-				$res .= $key . " = " . (is_numeric($value) ? $value : '"' . $value . '"') . "\n";
+				if ($isBaseLevel)
```

Comment:
looks like this flag is not really needed , if $baseKey has a value do one thing , else do another.